platforms/core-execution/file-watching/src/main/java/org/gradle/internal/watch/registry/FileWatcherRegistry.java

```diff
@@ -54,7 +55,7 @@ enum Type {
      *
      * @throws WatchingNotSupportedException when the native watchers can't be updated.
      */
-    void registerWatchableHierarchy(File watchableHierarchy, SnapshotHierarchy root);
+    void registerWatchableHierarchy(File watchableHierarchy, SnapshotHierarchy root, @Nullable File probeLocation);
```

Comment:
:x: Let's document what the probe location means, especially what a `null` value means.